src/core_codemods/lazy_logging.py

```diff
@@ -0,0 +1,160 @@
+                    return updated_node
+        return updated_node.with_changes(args=first_arg + new_args + remaining_args)
+
+    def make_args_for_plus(self, binop: cst.BinaryOperation) -> Optional[list[cst.Arg]]:
+        if self.is_str_concat(binop):
+            # Do not change explicit str concat, e.g.: `logging.info("one" + "two")
+            return None
+
+        if isinstance(binop.left, cst.SimpleString) and "%" in binop.left.value:
+            # Do no change `logging.info("Something: %s " + var)` since intention is unclear
+            return None
+        left_type = infer_expression_type(self.resolve_expression(binop.left))
+        right_type = infer_expression_type(self.resolve_expression(binop.right))
+        if left_type != right_type or (type_both_sides := left_type) not in {
+            BaseType.STRING,
+            BaseType.BYTES,
+        }:
+            # Cannot concat different types.
+            # Skip logging ints, etc. Eg: `logging.info(2+2)`
+            return None
+
+        format_strings, format_args = self.process_concat(binop)
+        combined_format_string = cst.SimpleString(
+            value=f"""{'"' if type_both_sides == BaseType.STRING else ""}{"".join(format_strings)}\""""
+        )
+        return [cst.Arg(value=combined_format_string)] + format_args
+
+    def make_args_for_modulo(self, binop: cst.BinaryOperation) -> list[cst.Arg]:
+        format_string = binop.left
+        format_args = binop.right
+        new_args = [cst.Arg(value=format_string)]
+        match format_args:
+            case cst.Tuple():
+                for element in format_args.elements:
+                    new_args.append(cst.Arg(value=element.value))
+            case _:
+                new_args.append(cst.Arg(value=format_args))
+        return new_args
+
+    def all_operators(self, node: cst.BinaryOperation):
+        if not isinstance(node.left, cst.BinaryOperation):
+            return [node.operator.__class__]
+        return [node.operator.__class__] + self.all_operators(node.left)
+
+    def is_str_concat(self, node: cst.CSTNode) -> bool:
+        match node:
+            case cst.BinaryOperation(operator=cst.Add()):
+                return self.is_str_concat(node.left) and self.is_str_concat(node.right)
+        return isinstance(node, cst.SimpleString)
+
+    def process_concat(
+        self, node: cst.CSTNode, format_strings=None, format_args=None
+    ) -> tuple[list[str], list[cst.Arg]]:
+        if format_strings is None:
+            format_strings = []
+        if format_args is None:
+            format_args = []
+
+        match node:
+            case cst.BinaryOperation(operator=cst.Add()):
+                self.process_concat(node.left, format_strings, format_args)
+                self.process_concat(node.right, format_strings, format_args)
+            case cst.SimpleString():
+                format_strings.append(node.value.strip("\"'"))
```

Comment:
I think this won't work if the string itself has any prefix (e.g. `u'some string'`). You can use `raw_value` to remove the quotes and prefix from the value of a `SimpleString`.

However I think you should try to preserve the string here with it's prefixes. Consider, for example:
```python
logger.info("message " + r"I\'m a raw value")
```
If you strip the `r` prefix the value of the string will be changed.